src/Query/Transformer/NamedArrayParameter.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Linio\Component\Database\Query\Transformer;
+
+use Linio\Component\Database\Exception\InvalidQueryException;
+use Linio\Component\Database\Query\Transformer;
+
+class NamedArrayParameter implements Transformer
+{
+    public function execute(string &$query, array &$params = []): void
+    {
+        foreach ($params as $paramKey => $paramValue) {
+            $isNumeric = is_numeric($paramKey);
+            $isArray = is_array($paramValue);
+
+            if ($isNumeric && $isArray) {
+                throw new InvalidQueryException(sprintf('Unnamed parameter "%s" can\'t have array value.', $paramKey));
+            }
+
+            if ($isNumeric || !$isArray) {
+                continue;
+            }
+
+            $placeholders = $this->placeholders($paramKey, $paramValue);
+            $params += $placeholders;
+
+            unset($params[$paramKey]);
+
+            $paramNames = array_keys($placeholders);
+
+            $hasDoubleColon = $paramKey[0] === ':';
```

Comment:
A double colon is '::', this is checking if the first character is a colon.

I would rename this to $startsWithColon to be more clear.

Reply:
you're right, thanks!